### DISCUSSION TITLE
Learning what can I use for Pydantic (version specific docs)

### DISCUSSION THREAD

**Discussion by AloisMahdal** [pydantic/pydantic#5532] [Question] [answered]
In past few weeks I've fell in love with Pydantic only to find out (**twice** in past week) that some features just randomly, silently, don't work with older versions (like `exclude` for `Field`, and now I'm in doubt about `use_tuple_values`).

Sure, that's 100% normal but usually there's a way docs
 *  inform about "added in version X.Y" right in the documentation (like, say, official Python library docs),
 *  provide easy way to switch to specific version when viewing the docs.

I know I can always read the changelogs or git-tagged sources but that does not seem consistent with otherwise great experience with Pydantic.

Perhaps this should be a bug report, but let's make it a question first: What is the best way to learn Pydantic, given that I'm limited to a specific version but still knowing what features can I use?

PS: I'm still in love with Pydantic :)

**Comment by samuelcolvin** [comment, marked as answer]
Thanks for the kind words.

Why do you need to change version? Most users don't go backwards in versions so this isn't a problem.

The short answer is that the docs aren't perfect, the project want managed by a team or anyone full time until recently.

We're aiming to improve this in V2.

**Reply by AloisMahdal** [reply]
Thank you for your fast response.

I'd love to use the newest version but it's limitation of the project I'm working on; we have version lock on 1.8.2.  It's a little bit of a "dependency hell" situation here but it's not something I can change easily, at least not until I finish what's in my focus right now.

I'll do my best to keep up and I will be looking forward for V2.

Thanks again!
